xradar/georeference/transforms.py

```diff
@@ -149,20 +153,26 @@ def get_x_y_z(ds, earth_radius=None, effective_radius_fraction=None):
     .. [1] Doviak and Zrnić, Doppler Radar and Weather Observations, Second
         Edition, 1993, p. 21.
     """
+
+    ds = ds.copy()
+
+    crs = radar_crs(ds)
+
     if earth_radius is None:
-        crs = get_crs(ds)
         earth_radius = get_earth_radius(crs, ds.latitude.values)
-        ds = ds.pipe(add_crs, crs)
 
     # Calculate x, y, and z from the dataset
-    ds["x"], ds["y"], ds["z"] = antenna_to_cartesian(
+    x, y, z = antenna_to_cartesian(
         ds.range,
         ds.azimuth,
         ds.elevation,
         earth_radius=earth_radius,
         effective_radius_fraction=effective_radius_fraction,
         site_altitude=ds.altitude.values,
     )
+    ds["x"] = x
```

Comment:
If we do it like this, maybe add the attributes from below here, too.

Reply:
I improved the handling.